src/services/analytics/TagManager.ts

```diff
@@ -38,40 +34,58 @@ const TagManager = {
 
     return script
   },
-  isInitialized: () => {
-    const GTM_SCRIPT = 'https://www.googletagmanager.com/gtm.js'
 
-    return !!document.querySelector(`[src^="${GTM_SCRIPT}"]`)
+  dataLayer: (data: Record<string, any>) => {
+    window[DATA_LAYER_NAME] = window[DATA_LAYER_NAME] || []
+    window[DATA_LAYER_NAME].push(data)
+
+    if (!IS_PRODUCTION) {
+      console.info('[GTM] -', data)
+    }
   },
+
   initialize: (args: TagManagerArgs) => {
-    if (TagManager.isInitialized()) {
-      return
+    window[DATA_LAYER_NAME] = window[DATA_LAYER_NAME] || []
+    // This function MUST be in `window`, otherwise GTM Consent Mode just doesn't work
+    ;(window as any).gtag = function () {
+      window[DATA_LAYER_NAME].push(arguments)
     }
```

Comment:
I don't understand why we have to overwrite the whole `gtag` object with a custom function instead of checking that window.gtag exists and if it doesn't we return early.

Reply:
Why would it exist if nobody defines it except us?

Reply:
I would assume the gtag script we load attaches it to the window.

Reply:
It's usually part of the code snippet because apparently it needs to exist before you load the gtm script. In our case it wasn't there. I don't know if we customized it or it was like this to begin with. Anyway, the gtag function wasn't there.

Reply:
According to [this](https://developers.google.com/tag-platform/gtagjs/install) we need to define that function ourselves indeed. Can we add a reference link in our code in case this comes up again?